kube/src/api/typed.rs

```diff
@@ -29,34 +29,41 @@ pub struct Api<K> {
     pub(crate) phantom: PhantomData<K>,
 }
 
+impl<K> Api<K> {
+    /// Converts an untyped and dynamic `Resource` into a typed `Api<K>`. j
+    ///
+    /// Note: No verification is performed of whether `K` matches the `Resource`.
+    /// It's fully possible to use this to try to read `ReplicaSet`s as `StatefulSet`s,
+    /// but the results will vary between useless and wildly confusing.
+    /// If you're not trying to reimplement `kubectl` then you'll probably want to use
+    /// `all` or `namespaced` instead.
+    pub fn from_dynamic_resource(client: Client, resource: Resource) -> Self {
```

Comment:
Given that this already exists in the form of `CustomResource`, maybe we can just have a `Resource::custom -> CrBuilder` function here instead to tie them together?